py3status/modules/uptime.py

```diff
@@ -0,0 +1,152 @@
+# -*- coding: utf-8 -*-
+"""
+Display system uptime.
+
+Configuration parameters:
+    default_format: start uptime in this format: 'pretty', 'short', or 'tiny'
+        (default 'pretty')
+    format: display format for this module (default 'up {format_uptime}')
+    format_pretty: show uptime in pretty format
+        *(default '[\?if=W {W} weeks, ][\?if=D {D} days, ]
+        [\?if=H {H} hours, ][\?if=M {M} minutes, ]')*
+    format_short: show uptime in short format
+        (default '[\?if=W {W}w ][\?if=D {D}d ][\?if=H {H}h ][\?if=M {M}m ]')
+    format_tiny: show uptime in tiny format
+        (default '[\?if=D {D}, ][\?if=H {H}:][\?if=M {M}:]')
+
+Datetime placeholders:
+    {G} Number of decades
+    {Y} Number of years
+    {W} Number of weeks
+    {D} Number of days
+    {H} Number of hours
+    {M} Number of minutes
+    {S} Number of seconds
+
+Format placeholders:
+    {format_uptime} uptime in multiple formats
+        To remove time clicking fun, remove {format_uptime}
+        and update format with desired datetime placeholders above.
+
+Format_* placeholders:
+    See datetime placeholders above.
+
+Note: If you don't use one of the placeholders, the value will be carried over
+to the next unit! For example, given an uptime of 1h30min:
+    If you use {minutes} as your only placeholder, then its value will be 90.
+    If you use {hours} and {minutes}, then its values will be 1 and 30, respectively.
+
+@author Alexis "Horgix" Chotard <alexis.horgix.chotard@gmail.com>
+@license BSD
+
+SAMPLE OUTPUT
+{'full_text': 'up 1 days, 18 hours, 20 minutes'}
+
+short
+{'full_text': 'up 1d 18h 20m'}
+
+tiny
+{'full_text': 'up 1, 18:20'}
+"""
+
+from time import time
+
+
+class Py3status:
+    """
+    """
+    # available configuration parameters
+    default_format = 'pretty'
+    format = 'up {format_uptime}'
+    format_pretty = '[\?if=W {W} weeks, ][\?if=D {D} days, ]' + \
+        '[\?if=H {H} hours, ][\?if=M {M} minutes, ]'
+    format_short = '[\?if=W {W}w ][\?if=D {D}d ][\?if=H {H}h ][\?if=M {M}m ]'
+    format_tiny = '[\?if=D {D}, ][\?if=H {H}:][\?if=M {M}:]'
```

Comment:
Then we end up with `0 weeks, 0 days, 0 hours, 38 minutes`. It gets old really fast.
I'm using what I believe is a sensible format so people can just add `uptime` and be happy with the format.